core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -219,23 +240,85 @@ private void resetInputAndInternalAndSeekToEndIntermediateTopicOffsets() {
+
+            if (!dryRun) {
+                for (final TopicPartition p : partitions) {
+                    client.position(p);
+                }
+                client.commitSync();
             }
-            if (intermediateTopicPartitions.size() > 0) {
-                client.seekToEnd(intermediateTopicPartitions);
+
+            if (notFoundInputTopics.size() > 0) {
+                System.out.println("Following input topics are not found, skipping them");
+                for (String topic : notFoundInputTopics) {
+                    System.out.println("Topic - " + topic);
+                }
             }
 
-            for (final TopicPartition p : partitions) {
-                client.position(p);
+            if (notFoundIntermediateTopics.size() > 0) {
+                System.out.println("Following intermediate topics are not found, skipping them");
+                for (String topic : notFoundIntermediateTopics) {
+                    System.out.println("Topic - " + topic);
+                }
             }
-            client.commitSync();
+            client.close();
+
         } catch (final RuntimeException e) {
             System.err.println("ERROR: Resetting offsets failed.");
             throw e;
         }
 
-        System.out.println("Done.");
+        if (!dryRun) {
+            System.out.println("Done.");
+        }
+    }
+
+    private void maybeSeekToEnd(KafkaConsumer<byte[], byte[]> client, Set<TopicPartition> intermediateTopicPartitions) {
+
+        final Boolean dryRun = options.has(dryRunOption);
+        final String groupId = options.valueOf(applicationIdOption);
+        final List<String> intermediateTopics = options.valuesOf(intermediateTopicsOption);
+
+        if (intermediateTopicPartitions.size() > 0) {
+            if (!dryRun) {
+                client.seekToEnd(intermediateTopicPartitions);
+            } else {
+                System.out.println("Following intermediate topics offsets will be reset to end (for consumer group " + groupId + ")");
+                for (String topic : intermediateTopics) {
+                    if (allTopics.contains(topic)) {
+                        System.out.println("Topic - " + topic);
+                    }
+                }
+            }
+        }
+
+    }
+
+    private void maybeSeekToBeginning(KafkaConsumer<byte[], byte[]> client, Set<TopicPartition> inputAndInternalTopicPartitions, List<String> internalTopics) {
+
+        final Boolean dryRun = options.has(dryRunOption);
+        final List<String> inputTopics = options.valuesOf(inputTopicsOption);
+        final String groupId = options.valueOf(applicationIdOption);
+
+        if (inputAndInternalTopicPartitions.size() > 0) {
+            if (!dryRun) {
+                client.seekToBeginning(inputAndInternalTopicPartitions);
+            } else {
+                System.out.println("Following input topics offsets will be reset to beginning (for consumer group " + groupId + ")");
```

Comment:
I understand the motivation and agree it's a good idea. It's only that 

`"Deleting all internal/auto-created topics for application <app.id>."`
and
`"Following internal/auto-created topics will be Deleted for application <app.id>."`

sound super similar. So not sure if we gain much. But it's quite subjective. It's ok if you leave as is. Please fix the typo though `Deleted` -> `deleted`.